syntax="proto3";
package snail.stream;

import "common.proto";

enum ExtentnodeMsgType {
    WRITE_EXTENT_REQ = 0;
    WRITE_EXTENT_RESP = 1;

    READ_EXTENT_REQ = 2;
    READ_EXTENT_RESP = 3;

    CREATE_EXTENT_REQ = 4;
    CREATE_EXTENT_RESP = 5;

    DELETE_EXTENT_REQ = 6; 
    DELETE_EXTENT_RESP = 7; 

    GET_EXTENT_REQ = 8; 
    GET_EXTENT_RESP = 9; 

    UPDATE_DISK_STATUS_REQ = 10;
    UPDATE_DISK_STATUS_RESP = 11;
}

message WriteExtentReq {
    CommonReq header = 1;
    uint32 diskid = 2;
    string extent_id = 3;
    uint64 off = 4;
    uint64 len = 5;  // exclude crc
}

message ReadExtentReq {
    CommonReq header = 1;
    uint32 diskid = 2;
    string extent_id = 3;
    uint64 off = 4;
    uint64 len = 5; // exclude crc
}

message ReadExtentResp{
    CommonResp header = 1;
    uint64 len = 2;  // exclude crc
}

message DeleteExtentReq {
    CommonResp header = 1;
    uint32 diskid = 2;
    string extent_id = 3;
}

message CreateExtentReq {
    CommonResp header = 1;
    uint32 diskid = 2;
    string extent_id = 3;
}

message GetExtentReq {
    CommonResp header = 1;
    uint32 diskid = 2;
    string extent_id = 3;
}

message GetExtentResp {
    CommonResp header = 1;
    uint64 len = 2;
    uint32 ctime = 3;
}

message UpdateDiskStatusReq {
    CommonResp header = 1;
    uint32 diskid = 2;
    uint32 status = 3;
}
